### PR TITLE
Fix a number of compiler warnings

### PR DESCRIPTION
**Pull request by glaubitz**
- Fix incorrect format specifiers when calling sprintf()
- Fix potentially uninitialized variable
- Remove unused variables

### CODE REVIEW

File: docs/versionhistory.rst [hunk 1]
```diff
@@ -8,6 +8,7 @@ This library adheres to `Semantic Versioning <http://semver.org/>`_.
 **UNRELEASED**
 
 - Fixed compilation of C extension failing on GCC 14
+- Fixed compiler warnings when building C extension
```
Comment (agronholm) [inline docs/versionhistory.rst:11]: Does this warrant a new changelog entry? Weren't these two PRs actually both to fix the compilation problem on GCC 14?

Comment (glaubitz) [reply docs/versionhistory.rst:11]: No, this PR fixes unrelated compiler warnings. The other own fixed an actual build error.

**Comment by agronholm** [comment]
Thanks!

**Comment by glaubitz** [comment]
Great, thanks. Hoping for a release soon so distributions currently switching to GCC 14 can drop the local patches.

**Comment by agronholm** [comment]
v5.6.4 is brewing right now: https://github.com/agronholm/cbor2/actions/runs/9397794316

**Comment by glaubitz** [comment]
> v5.6.4 is brewing right now: https://github.com/agronholm/cbor2/actions/runs/9397794316

Awesome, thanks. I'll update it in openSUSE as soon it's out!

**Pull request merged**